test/groups/test_unitary_group.jl

```diff
@@ -1,43 +1,73 @@
 using LieGroupsTestSuite
 
 @testset "Unitary Group" begin
-    G = UnitaryGroup(2)
-    g1 = 1 / sqrt(2) * ComplexF64[1.0 1.0; -1.0 1.0]
-    g2 = ComplexF64[0.0 -1.0; 1.0 0.0]
-    g3 = ComplexF64[-1.0 0.0; 0.0 1.0] #one with det -1
-    X1 = [0.0 1.0im; 1.0im 0.0]
-    X2 = ComplexF64[0.0 1.0; -1.0 0.0]
-    X3 = ComplexF64[0.0 -0.5; 0.5 0.0]
-    properties = Dict(
-        :Name => "The unitary group",
-        :Points => [g1, g2, g3],
-        :Vectors => [X1, X2, X3],
-        :Rng => Random.MersenneTwister(),
-        :Functions => [
-            adjoint,
-            compose,
-            conjugate,
-            diff_inv,
-            diff_left_compose,
-            diff_right_compose,
-            exp,
-            # hat,
-            identity_element,
-            inv,
-            inv_left_compose,
-            inv_right_compose,
-            is_identity,
-            lie_bracket,
-            log,
-            rand,
-            show,
-            # vee,
-        ],
-    )
-    expectations = Dict(:repr => "UnitaryGroup(2)")
-    test_lie_group(G, properties, expectations)
+    @testset "Complex Unitary group" begin
+        G = UnitaryGroup(2)
+        g1 = 1 / sqrt(2) * ComplexF64[1.0 1.0; -1.0 1.0]
+        g2 = ComplexF64[0.0 -1.0; 1.0 0.0]
+        g3 = ComplexF64[-1.0 0.0; 0.0 1.0] #one with determinant -1
+        X1 = [0.0 1.0im; 1.0im 0.0]
+        X2 = ComplexF64[0.0 1.0; -1.0 0.0]
+        X3 = ComplexF64[0.0 -0.5; 0.5 0.0]
+        properties = Dict(
+            :Name => "The unitary group",
+            :Points => [g1, g2, g3],
+            :Vectors => [X1, X2, X3],
+            :Rng => Random.MersenneTwister(),
+            :Functions => [
+                adjoint,
+                compose,
+                conjugate,
+                diff_inv,
+                diff_left_compose,
+                diff_right_compose,
+                exp,
+                # hat,
+                identity_element,
+                inv,
+                inv_left_compose,
+                inv_right_compose,
+                is_identity,
+                lie_bracket,
+                log,
+                rand,
+                show,
+                # vee,
+            ],
+        )
+        expectations = Dict(:repr => "UnitaryGroup(2)")
+        test_lie_group(G, properties, expectations)
+        G2 = UnitaryGroup(2; parameter=:field)
+        expectations2 = Dict(:repr => "UnitaryGroup(2; parameter=:field)")
+        test_lie_group(G2, properties, expectations2)
+    end
+    @testset "Quaternion Unitary group" begin
+        G = UnitaryGroup(1, ℍ)
+        g1 = quat(1.0)
+        g2 = quat(0.0, 1.0, 0.0, 0.0)
+        g3 = quat(0.0, 0.0, 1.0, 0.0)
+        properties = Dict(
+            :Name => "The quaternion unitary group",
+            :Points => [g1, g2, g3],
+            :Vectors => [quat(0.0)],
+            :Mutating => false,
+            :Rng => Random.MersenneTwister(),
+            :Functions => [rand, exp, log, show],
```

Comment:
Could you also test `compose`, `inv`, `exp_fused` and the mutating case? Manifolds.jl currently seems to support all of these.

Reply:
^exp_fused` does not yet exist in LieGroups.jl – and by now I think it will take quite a time until it does, if _I_ should implement that, not this year, since I am loosing a bit of motivation here. I am growing a bit of the feeling that I just wasted half a year of programming on something no one will ever use with this package.

I can check for compose and inv but I am not sure how to approach the mutating case?

Reply:
OK, then let's leave `exp_fused` out for now. The mutating case should work the same way as in Manifolds.jl -- the quaternion is put into `Ref` or one-element array with `fill`.

Reply:
Sure, can try the zero-array/fill thing. At least add the tests today and if they fill try to find time to fix that throughout the week

Reply:
Hm, well, compose and inv also do not work. Ok, Then this “ah this is easy” PR will also turn into a few weeks of work I fear.

I really wish at some point I am able to finish any PR in a finite amount of time again.